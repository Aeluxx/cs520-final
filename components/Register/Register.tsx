// Copied from
// https://github.com/mui-org/material-ui/blob/next/docs/src/pages/getting-started/templates/sign-in/SignIn.tsx

import Button from '@mui/material/Button'
import CssBaseline from '@mui/material/CssBaseline'
import TextField from '@mui/material/TextField'
import FormControlLabel from '@mui/material/FormControlLabel'
import Checkbox from '@mui/material/Checkbox'
import Link from '@mui/material/Link'
import Grid from '@mui/material/Grid'
import Box from '@mui/material/Box'
import Typography from '@mui/material/Typography'
import Container from '@mui/material/Container'
import { FormEvent, useState } from 'react'
import axios from 'axios'
import { useRouter } from 'next/router'

// https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript
function validateEmail(email: any) {
  const re =
    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
  return re.test(String(email).toLowerCase())
}

const Register = () => {
  const router = useRouter()
  const [errMessage, setErrMessage] = useState<string>('')
  const [name, setName] = useState('')
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [confirmPassword, setConfirmPassword] = useState('')

  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault()
    const res = await axios.post('http://localhost:3000/api/register', {
      name,
      email,
      password,
    }).then(res => router.push('/login')).catch(() => setErrMessage('Please make sure all fields are filled out...'))

  }

  return (
    <Container component='main' maxWidth='xs'>
      <CssBaseline />
      <Box
        sx={{
          marginTop: 6,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
        }}>
        <Typography component='h1' variant='h5'>
          Register
        </Typography>
        <Box component='form' onSubmit={handleSubmit} noValidate sx={{ mt: 1 }}>
          <TextField
            margin='normal'
            value={name}
            onChange={e => setName(e.target.value)}
            required
            fullWidth
            id='username'
            label='Username'
            name='username'
            autoFocus
          />
          <TextField
            margin='normal'
            value={email}
            onChange={e => setEmail(e.target.value)}
            required
            fullWidth
            id='email'
            label='Email Address'
            name='email'
          />
          <TextField
            margin='normal'
            value={password}
            onChange={e => setPassword(e.target.value)}
            required
            fullWidth
            label='Password'
            type='password'
          />
          <TextField
            margin='normal'
            value={confirmPassword}
            onChange={e => setConfirmPassword(e.target.value)}
            required
            fullWidth
            label='Confirm Password'
            type='password'
          />
          <FormControlLabel control={<Checkbox value='remember' color='primary' />} label='Remember me' name='rememberMe' id='rememberMe' />
          <Button type='submit' fullWidth variant='contained' sx={{ mt: 3, mb: 2 }}>
            Register
          </Button>
        </Box>
      </Box>

      <Typography variant='body2'>
        Don&apos;t have an account? <Link href='/register'>Create an account</Link>
      </Typography>
      <Typography variant='caption' color='red'>
        {errMessage}
      </Typography>
    </Container>
  )
}

export default Register
