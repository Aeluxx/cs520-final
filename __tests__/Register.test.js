import React from 'react';
import '@testing-library/jest-dom';
import Register from '../components/Register/Register';
import ShallowRenderer from 'react-test-renderer/shallow';

let realUseContext;
let useContextMock;

beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
});

afterEach(() => {
    React.useContext = realUseContext;
});

it("Register renders without error", () => {
    useContextMock.mockReturnValue("Test Value");
    const shallow = new ShallowRenderer();
    const register = shallow.render(<Register />);
});

it("Register matches Snapshot", () => {
    useContextMock.mockReturnValue("Test Value");
    const shallow = new ShallowRenderer();
    const register = shallow.render(<Register />);
    const result = shallow.getRenderOutput();
    expect(result).toMatchSnapshot();
});