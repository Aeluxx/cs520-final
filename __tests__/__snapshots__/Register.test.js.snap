// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register matches Snapshot 1`] = `
<ForwardRef(Container)
  component="main"
  maxWidth="xs"
>
  <CssBaseline />
  <ForwardRef(Box)
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "marginTop": 6,
      }
    }
  >
    <ForwardRef(Typography)
      component="h1"
      variant="h5"
    >
      Register
    </ForwardRef(Typography)>
    <ForwardRef(Box)
      component="form"
      noValidate={true}
      onSubmit={[Function]}
      sx={
        Object {
          "mt": 1,
        }
      }
    >
      <ForwardRef(TextField)
        autoFocus={true}
        fullWidth={true}
        id="username"
        label="Username"
        margin="normal"
        name="username"
        onChange={[Function]}
        required={true}
        value=""
      />
      <ForwardRef(TextField)
        fullWidth={true}
        id="email"
        label="Email Address"
        margin="normal"
        name="email"
        onChange={[Function]}
        required={true}
        value=""
      />
      <ForwardRef(TextField)
        fullWidth={true}
        label="Password"
        margin="normal"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
      <ForwardRef(TextField)
        fullWidth={true}
        label="Confirm Password"
        margin="normal"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
      <ForwardRef(FormControlLabel)
        control={
          <ForwardRef(Checkbox)
            color="primary"
            value="remember"
          />
        }
        id="rememberMe"
        label="Remember me"
        name="rememberMe"
      />
      <ForwardRef(Button)
        fullWidth={true}
        sx={
          Object {
            "mb": 2,
            "mt": 3,
          }
        }
        type="submit"
        variant="contained"
      >
        Register
      </ForwardRef(Button)>
    </ForwardRef(Box)>
  </ForwardRef(Box)>
  <ForwardRef(Typography)
    variant="body2"
  >
    Don't have an account? 
    <ForwardRef(Link)
      href="/register"
    >
      Create an account
    </ForwardRef(Link)>
  </ForwardRef(Typography)>
  <ForwardRef(Typography)
    color="red"
    variant="caption"
  >
    
  </ForwardRef(Typography)>
</ForwardRef(Container)>
`;
